components/console/single_command_tool.rst

```diff
@@ -6,68 +6,51 @@ Building a single Command Application
-            return 'my_command';
-        }
-
-        /**
-         * Gets the default commands that should always be available.
-         *
-         * @return array An array of default Command instances
-         */
-        protected function getDefaultCommands()
-        {
-            // Keep the core default commands to have the HelpCommand
-            // which is used when using the --help option
-            $defaultCommands = parent::getDefaultCommands();
-
-            $defaultCommands[] = new MyCommand();
-
-            return $defaultCommands;
-        }
-
-        /**
-         * Overridden so that the application doesn't expect the command
-         * name to be the first argument.
-         */
-        public function getDefinition()
-        {
-            $inputDefinition = parent::getDefinition();
-            // clear out the normal first argument, which is the command name
-            $inputDefinition->setArguments();
-
-            return $inputDefinition;
-        }
-    }
-
-When calling your console script, the command ``MyCommand`` will then always
-be used, without having to pass its name.
-
-You can also simplify how you execute the application::
-
-    #!/usr/bin/env php
-    <?php
-    // command.php
-
-    use Acme\Tool\MyApplication;
-
-    $application = new MyApplication();
-    $application->run();
+it is possible to remove this need declaring a single command application::
+
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
+
+  use Symfony\Component\Console\Application;
+  use Symfony\Component\Console\Input\InputArgument;
+  use Symfony\Component\Console\Input\InputInterface;
+  use Symfony\Component\Console\Input\InputOption;
+  use Symfony\Component\Console\Output\OutputInterface;
+
+  (new Application('echo', '1.0.0'))
+    ->register('echo')
+        ->addArgument('foo', InputArgument::OPTIONAL, 'The directory')
+        ->addOption('bar', null, InputOption::VALUE_REQUIRED)
+        ->setCode(function(InputInterface $input, OutputInterface $output) {
+            // output arguments and options
+        })
+    ->getApplication()
+    ->setDefaultCommand('echo', true) // Single command application
+    ->run();
+
+:method:`Symfony\\Component\\Console\\Application::setDefaultCommand` accepts a
+boolean as a second parameter. If true, the command ``echo``will then always be
+used, without having to pass its name.
+
+Of course, you can still register a command as usual::
+
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
+
+  use Symfony\Component\Console\Application;
+  use Symfony\Component\Console\Input\InputArgument;
+  use Symfony\Component\Console\Input\InputInterface;
+  use Symfony\Component\Console\Input\InputOption;
+  use Symfony\Component\Console\Output\OutputInterface;
+
+  use Acme\Command\DefaultCommand;
+
+  $application = new Application('echo', '1.0.0');
+  $command = new DefaultCommand();
+
```

Comment:
Just below, on the PR checks section you can click on "details" of the platformsh check